array-filter/filter.js

```diff
@@ -0,0 +1,7 @@
+function filter(array, predicate) {
+  var tempArr = [];
+  for (let i = 0; i < array.length; i++) {
+    tempArr.push(predicate(array[i]));
+  }
+  return tempArr;
+}
```

Comment:
Good attempt on the bonus challenge, but this won't quite work right. You need an `if` statement within the loop to determine if the current item should be added to the `tempArr`.